deluge/ui/console/modes/connectionmanager.py

```diff
@@ -127,12 +127,14 @@ def add_popup(self):
 
     def add_host(self, hostname, port, username, password):
         log.info('Adding host: %s', hostname)
+        if port.isdecimal():
+            port = int(port)
```

Comment:
This can be handled by replacing the `port`'s `add_text_input` in the `add_popup` method with:
```python
popup.add_int_spin_input('port', _('Port:'), min_val=0, max_val=65535, value=DEFAULT_PORT)
```